apm-lambda-extension/extension/route_handlers.go

```diff
@@ -29,68 +29,73 @@ type AgentData struct {
 }
 
 // URL: http://server/
-func handleInfoRequest(handler *serverHandler, w http.ResponseWriter, r *http.Request) {
-	client := &http.Client{}
+func handleInfoRequest(apmServerUrl string) func(w http.ResponseWriter, r *http.Request) {
+	return func(w http.ResponseWriter, r *http.Request) {
+		client := &http.Client{}
 
-	req, err := http.NewRequest(r.Method, handler.config.apmServerUrl, nil)
-	//forward every header received
-	for name, values := range r.Header {
-		// Loop over all values for the name.
-		for _, value := range values {
-			req.Header.Set(name, value)
+		req, err := http.NewRequest(r.Method, apmServerUrl, nil)
+		//forward every header received
+		for name, values := range r.Header {
+			// Loop over all values for the name.
+			for _, value := range values {
+				req.Header.Set(name, value)
+			}
+		}
+		if err != nil {
+			log.Printf("could not create request object for %s:%s: %v", r.Method, apmServerUrl, err)
+			return
 		}
-	}
-	if err != nil {
-		log.Printf("could not create request object for %s:%s: %v", r.Method, handler.config.apmServerUrl, err)
-		return
-	}
 
-	resp, err := client.Do(req)
-	if err != nil {
-		log.Printf("error forwarding info request (`/`) to APM Server: %v", err)
-		return
-	}
-	defer resp.Body.Close()
-	body, err := ioutil.ReadAll(resp.Body)
-	if err != nil {
-		log.Printf("could not read info request response to APM Server: %v", err)
-		return
-	}
+		resp, err := client.Do(req)
+		if err != nil {
+			log.Printf("error forwarding info request (`/`) to APM Server: %v", err)
+			return
+		}
+		defer resp.Body.Close()
+		body, err := ioutil.ReadAll(resp.Body)
```

Comment:
The `body` shouldn't be massive in size, but a good habit is to operate on the `resp.Body` as an `io.Reader`, and use something like `io.Copy` to keep memory allocation down.

Reply:
thanks, I'll change that 👍🏼